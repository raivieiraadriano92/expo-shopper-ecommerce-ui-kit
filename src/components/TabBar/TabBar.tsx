import { FunctionComponent } from 'react'

import { Box, HStack, IconButton } from 'native-base'

import Icons from '../Icons'

export type TabBarKeyItem = 'home' | 'search' | 'cart' | 'order' | 'account'

type TabBarItem = {
  key: TabBarKeyItem
  icon: keyof typeof Icons
}

type Props = {
  activeKey: TabBarKeyItem
  onPress(key: TabBarKeyItem): void
}

const items: TabBarItem[] = [
  { key: 'home', icon: 'Home' },
  { key: 'search', icon: 'Search' },
  { key: 'cart', icon: 'Cart' },
  { key: 'order', icon: 'List' },
  { key: 'account', icon: 'User' }
]

export const TAB_BAR_HEIGHT_TOKEN = 16

export const TabBar: FunctionComponent<Props> = ({ activeKey, onPress }) => (
  <Box
    bg="white"
    borderTopRadius="3xl"
    safeAreaBottom
    style={{
      shadowColor: '#1b1956',
      shadowOffset: {
        width: 4,
        height: 0
      },
      shadowOpacity: 0.1,
      shadowRadius: 30,
      elevation: 24
    }}>
    <HStack alignItems="center" h={TAB_BAR_HEIGHT_TOKEN} justifyContent="space-between" px={4}>
      {items.map((item) => {
        const IconComponent = Icons[item.icon]

        return (
          <IconButton
            {...(activeKey === item.key ? { bg: 'primary.500' } : {})}
            icon={<IconComponent {...(activeKey === item.key ? { color: 'white' } : {})} />}
            key={item.key}
            onPress={() => onPress(item.key)}
            size="sm"
            testID={`TabBarItem.${item.key}`}
            _hover={{
              bg: 'primary.500',
              _icon: { color: 'white' }
            }}
            _pressed={{
              bg: 'primary.500',
              _icon: { color: 'white' }
            }}
          />
        )
      })}
    </HStack>
  </Box>
)
