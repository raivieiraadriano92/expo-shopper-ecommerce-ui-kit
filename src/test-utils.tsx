import { FunctionComponent } from 'react'

import { render } from '@testing-library/react-native'

import { AppWrapper } from '../src/AppWrapper'

const Wrapper: FunctionComponent = ({ children }) => <AppWrapper>{children}</AppWrapper>

const customRender: typeof render = (ui, options) => render(ui, { wrapper: Wrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
