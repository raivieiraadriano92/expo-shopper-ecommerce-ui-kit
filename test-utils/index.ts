import { render } from '@testing-library/react-native'
import { AppWrapper } from 'src/AppWrapper'

const customRender: typeof render = (ui, options) => render(ui, { wrapper: AppWrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
