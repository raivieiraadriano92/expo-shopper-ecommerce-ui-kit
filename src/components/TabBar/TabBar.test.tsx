import { render, waitFor } from 'test-utils'

import { TabBar } from './TabBar'

describe('<TabBar />', () => {
  it('renders correctly', async () => {
    const { findByTestId, toJSON } = render(<TabBar activeKey="home" onPress={() => {}} />)

    await waitFor(() => findByTestId('TabBarItem.home'))

    expect(toJSON()).toMatchSnapshot()
  })
})
