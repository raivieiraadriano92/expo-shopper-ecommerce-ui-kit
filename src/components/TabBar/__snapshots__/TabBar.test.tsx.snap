// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBar /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
        },
        Object {
          "elevation": 24,
          "shadowColor": "#1b1956",
          "shadowOffset": Object {
            "height": 0,
            "width": 4,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 30,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <View
        _disabled={
          Object {
            "opacity": 0.5,
          }
        }
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#07BFA5",
            "borderRadius": 9999,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 4,
            "width": 40,
          }
        }
        testID="TabBarItem.home"
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          color={4294967295}
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke=""
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#FFFFFF",
                "height": 20,
                "width": 20,
              },
              Object {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          tintColor={4294967295}
          vbHeight={20}
          vbWidth={20}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M2.5 7.49996L10 1.66663L17.5 7.49996V16.6666C17.5 17.1087 17.3244 17.5326 17.0118 17.8451C16.6993 18.1577 16.2754 18.3333 15.8333 18.3333H4.16667C3.72464 18.3333 3.30072 18.1577 2.98816 17.8451C2.67559 17.5326 2.5 17.1087 2.5 16.6666V7.49996Z"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M7.5 18.3333V10H12.5V18.3333"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={Object {}}
      />
      <View
        _disabled={
          Object {
            "opacity": 0.5,
          }
        }
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 4,
            "width": 40,
          }
        }
        testID="TabBarItem.search"
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          color={4280690214}
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke=""
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#262626",
                "height": 20,
                "width": 20,
              },
              Object {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          tintColor={4280690214}
          vbHeight={20}
          vbWidth={20}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M9.16667 15.8333C12.8486 15.8333 15.8333 12.8486 15.8333 9.16667C15.8333 5.48477 12.8486 2.5 9.16667 2.5C5.48477 2.5 2.5 5.48477 2.5 9.16667C2.5 12.8486 5.48477 15.8333 9.16667 15.8333Z"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M17.5 17.5L13.875 13.875"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={Object {}}
      />
      <View
        _disabled={
          Object {
            "opacity": 0.5,
          }
        }
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 4,
            "width": 40,
          }
        }
        testID="TabBarItem.cart"
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          color={4280690214}
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke=""
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#262626",
                "height": 20,
                "width": 20,
              },
              Object {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          tintColor={4280690214}
          vbHeight={20}
          vbWidth={20}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M5 1.66675L2.5 5.00008V16.6667C2.5 17.1088 2.67559 17.5327 2.98816 17.8453C3.30072 18.1578 3.72464 18.3334 4.16667 18.3334H15.8333C16.2754 18.3334 16.6993 18.1578 17.0118 17.8453C17.3244 17.5327 17.5 17.1088 17.5 16.6667V5.00008L15 1.66675H5Z"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M2.5 5H17.5"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M13.3337 8.33325C13.3337 9.21731 12.9825 10.0652 12.3573 10.6903C11.7322 11.3154 10.8844 11.6666 10.0003 11.6666C9.11627 11.6666 8.26842 11.3154 7.6433 10.6903C7.01818 10.0652 6.66699 9.21731 6.66699 8.33325"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={Object {}}
      />
      <View
        _disabled={
          Object {
            "opacity": 0.5,
          }
        }
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 4,
            "width": 40,
          }
        }
        testID="TabBarItem.order"
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          color={4280690214}
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke=""
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#262626",
                "height": 20,
                "width": 20,
              },
              Object {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          tintColor={4280690214}
          vbHeight={20}
          vbWidth={20}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M17.5 8H2.5"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.5}
              />
              <RNSVGPath
                d="M17.5 4H2.5"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.5}
              />
              <RNSVGPath
                d="M17.5 12H2.5"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.5}
              />
              <RNSVGPath
                d="M17.5 16H2.5"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.5}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={Object {}}
      />
      <View
        _disabled={
          Object {
            "opacity": 0.5,
          }
        }
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "height": 40,
            "justifyContent": "center",
            "paddingBottom": 4,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 4,
            "width": 40,
          }
        }
        testID="TabBarItem.account"
      >
        <RNSVGSvgView
          accessibilityRole="image"
          align="xMidYMid"
          bbHeight={20}
          bbWidth={20}
          color={4280690214}
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke=""
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#262626",
                "height": 20,
                "width": 20,
              },
              Object {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          tintColor={4280690214}
          vbHeight={20}
          vbWidth={20}
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M16.6667 17.5V15.8333C16.6667 14.9493 16.3155 14.1014 15.6904 13.4763C15.0653 12.8512 14.2174 12.5 13.3334 12.5H6.66671C5.78265 12.5 4.93481 12.8512 4.30968 13.4763C3.68456 14.1014 3.33337 14.9493 3.33337 15.8333V17.5"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
              <RNSVGPath
                d="M9.99996 9.16667C11.8409 9.16667 13.3333 7.67428 13.3333 5.83333C13.3333 3.99238 11.8409 2.5 9.99996 2.5C8.15901 2.5 6.66663 3.99238 6.66663 5.83333C6.66663 7.67428 8.15901 9.16667 9.99996 9.16667Z"
                fill={null}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  Array [
                    2,
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={1.4}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
